docs/samples/storage/storageSpec/common_secret.yaml

```diff
@@ -0,0 +1,15 @@
+apiVersion: v1
+stringData:
+  localMinIO: |
+    {
+      "type": "s3",
```

Comment:
I think the storage type should be on the `storageSpec` ?

Reply:
In the original modelmesh design, users don't need to be aware of the storage type because the user management service are responsible for the credential creation (in Kubeflow would be the profile controller). We still allow users to override the type here using `storage.parameters.type`. @njhill are you aware of other reasons why modelmesh stays with this design?

Reply:
@yuzisun @Tomcli Yes that's right, the type is a parameter in the storage entries in the common secret that are referenced in the predictors via their key, so the type is generally redundant there (you could still name the storage key itself `s3` or `s3public1` or whatever). It can still be specified explicitly though as a parameter called "type", for example if you are specifying the storage completely without reference to a secret entry, or you can specify only the type in which case it will look for a storage entry named `default_<type>`.

I guess we could promote this to an optional field of the storage spec but I'm not sure we want to encourage its use. But I'm fine with it if you think we should.

Reply:
@njhill @Tomcli I am just thinking the public s3 bucket case where secret is not required, not sure how the storage spec can handle it as there is no storage type. 
```
apiVersion: serving.kserve.io/v1beta1
kind: InferenceService
metadata:
  name: example-sklearn-isvc
spec:
  predictor:
    sklearn:
      storage:
        path: sklearn # Model path inside the bucket
        # schemaPath: null # Optional schema files for payload schema
        parameters: # Parameters to override the default values inside the common secret.
          bucket: example-models
```

Reply:
There is also use case when user is using AWS IAM roles, secret is not used and you annotate the role name on service account.

Reply:
I updated the python server to also take `anonymous` as the parameters. The below definition should able to run anonymous mode for aws
```
apiVersion: serving.kserve.io/v1beta1
kind: InferenceService
metadata:
  name: example-sklearn-isvc
spec:
  predictor:
    sklearn:
      storage:
        path: sklearn # Model path inside the bucket
        parameters: # Parameters to override the default values inside the common secret.
          anonymous: True
```

Reply:
For AWS IAM roles, right now users probably have to migrant the iam secret credentials to the common secret. Since S3 could have different providers (e.g. aws, ibm cloud, minio), we can look into adding some extra reconcile logics for automate the migration of s3 secrets from different providers to the common secret.

Reply:
> @njhill @Tomcli I am just thinking the public s3 bucket case where secret is not required, not sure how the storage spec can handle it as there is no storage type.

@yuzisun @Tomcli sorry for delayed response, have been off sick and still recovering. The type can still be provided in the `parameters` map, so it should still work fine like this in the public s3 bucket case with no secret:
```yaml
apiVersion: serving.kserve.io/v1beta1
kind: InferenceService
metadata:
  name: example-sklearn-isvc
spec:
  predictor:
    sklearn:
      storage:
        path: sklearn # Model path inside the bucket
        parameters:
          type: s3
          bucket: example-models
```
As I discussed [above](https://github.com/kserve/kserve/pull/1899#discussion_r787283720) we could consider promoting this to an explicit field of the storage spec, but I'd vote against that since in _most_ cases it shouldn't be needed.

> There is also use case when user is using AWS IAM roles, secret is not used and you annotate the role name on service account.

@yuzisun part of the reason for the new approach was to standardize how the various storage parameters are configured for the various storage types, so yes like @Tomcli said the "new" way to do this would be to set these parameters in either an explicit entry in the common storage secret (referenced by key in the specs) or a default entry (which would give the same behaviour from the InfereneService spec pov). But you're right that this specific case would require values to be set in a secret when they weren't before - is that a problem though? IMO it's better for things to be consistent. Of course you could technically still avoid use of a secret by including those params explicitly in the `parameters` section of the storage spec.

As we had discussed before (and @Tomcli mentions above), we can have backwards-compatibility logic which populates/updates default entries in the new storage secret based on various different ways that the storage parameters are currently set (i.e. watches the both the service account(s) and associated secrets)